lib/utils.js

```diff
@@ -95,15 +139,33 @@ const utils = {
         utils.logError('Configuration file missing for this directory, run losant configure to generate this file.');
         return;
       }
-      config.file = file;
+      if (!config.apiUrl) {
+        const url = utils.getApiURL(userConfig);
+        if (url) {
+          config.apiUrl = url;
+          await writeFile(foundFile, yaml.safeDump(userConfig));
+        }
+      }
+      config.file = foundFile;
     }
     return config;
   },
 
   loadConfig: async (file) => {
-    const userConfig = await utils.loadUserConfig(file);
-    const appConfig = await utils.loadApplicationConfig(file);
-    return merge(userConfig, appConfig);
+    let userConfig = await utils.loadUserConfig();
+    const appConfig = await utils.loadApplicationConfig(file, userConfig);
+    if (!appConfig) { return {}; }
```

Comment:
don't you need to exit with an error message here, not return `{}`?

Reply:
as discussed in chat I need to restructure the exit flow of this. Currently, all the commands will exit if no applicationId or accessToken is returned to the command, expecting either one of the error messages to have already been printed. I have tests around this case. I'll open another task to rework this to make testing exit routes easier and clean up the extra code I have around this case.